Docs: adding information about options LOGIN and HASH into CREATE/ALTER USER (#14967)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/syntax/create-user.md

# CREATE USER

Creates a database user.

Syntax:

```yql
CREATE USER user_name [option]
```

{% include [!](../../../_includes/user-options.md) %}

{% include [!](../../../_includes/do-not-create-users-in-ldap.md) %}

> ## PASSWORD
> 
> The database administrator can set the user's password when creating it. If the option is not `PASSWORD NULL`, enclose password in quotation marks.
> 
> Examples:

Reviewer: ```suggestion
## PASSWORD

A user's password can be set during creation. If the option is not `PASSWORD NULL`, enclose the password in quotation marks.

Examples:  
```

```yql
CREATE USER user1 PASSWORD 'password';
```

```yql
CREATE USER user1 PASSWORD "password";
```

```yql
CREATE USER user1 PASSWORD NULL;
```

## HASH

{% include [!](../../../_includes/hash-option.md) %}

> So, the `HASH` option must get a JSON object with the following three fields:
> 
> * `hash` — hash value in base64 format;
> * `salt` — [salt](https://en.wikipedia.org/wiki/Salt_(cryptography)) in base64 format;
> * `type` – hashing algorithm (always set it to `argon2id`).

Reviewer: ```suggestion
```
Seems to be a duplicate of what's included on line 35.

Example:

```yql
CREATE USER user1 HASH '{
    "hash": "p4ffeMugohqyBwyckYCK1TjJfz3LIHbKiGL+t+oEhzw=",
    "salt": "U+tzBtgo06EBQCjlARA6Jg==",
    "type": "argon2id"
}';
```

> ## NOLOGIN
> 
> Database administrator can create user blocked from logging in to the database.

Reviewer: ```suggestion
## NOLOGIN

A newly created user can blocked from logging in to the database using this option. It is mutually exclusive with `LOGIN`.
```

Example:

```yql
CREATE USER user1 NOLOGIN;
```

## LOGIN

> The option explicitly indicates that the created user is unblocked. If the `LOGIN` and `NOLOGIN` options are not specified, new users are not blocked.

Reviewer: ```suggestion
This option explicitly indicates that the created user is unblocked. It is mutually exclusive with `NOLOGIN`.

If neither the `LOGIN` nor `NOLOGIN` option is specified, new users are **not** blocked.  
```
